### PR TITLE
bump protoc in samples to e.g. get support for optionals

### PR DESCRIPTION
**Pull request by sebastian-alfers**
https://typesafe.slack.com/archives/C01FDELQMGR/p1643193193198700

### CODE REVIEW

File: samples/java-customer-registry-quickstart/pom.xml [hunk 1]
```diff
@@ -70,7 +70,7 @@
               <includeSourceInfoInDescriptorSet>true</includeSourceInfoInDescriptorSet>
               <descriptorSetFileName>user-function.desc</descriptorSetFileName>
               <checkStaleness>true</checkStaleness>
-              <protocArtifact>com.google.protobuf:protoc:3.11.4:exe:${os.detected.classifier}</protocArtifact>
+              <protocArtifact>com.google.protobuf:protoc:3.15.0:exe:${os.detected.classifier}</protocArtifact>
```
Comment (johanandren) [inline samples/java-customer-registry-quickstart/pom.xml:73]: Better align it with Akka gRPC and use 3.19.2?

Comment (sebastian-alfers) [reply samples/java-customer-registry-quickstart/pom.xml:73]: Sounds good!

**Review by johanandren — APPROVED** [review]
LGTM!

**Review by johanandren — CHANGES_REQUESTED** [review]
Ups LGTM:ed too quick, we should update the maven archetypes as well:

maven-java/akkaserverless-maven-archetype-event-sourced-entity/src/main/resources/archetype-resources/pom.xml
and
maven-java/akkaserverless-maven-archetype-value-entity/src/main/resources/archetype-resources/pom.xml

**Comment by raboof** [comment]
> we should update the maven archetypes as well

would be nice to get something like https://github.com/akka/akka-grpc/blob/main/project/VersionSyncCheckPlugin.scala in here, but not urgent

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Review by octonato — APPROVED** [review]
LGTM

**Pull request merged**